tests/container-nesting

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+set -ex
+
+# Install and setup LXD.
+install_lxd
+lxd init --auto
+
+# Install dependencies
+install_deps jq
+
+poolDriverList="${1:-dir btrfs lvm zfs}"
+
+IMAGE="${TEST_IMG:-ubuntu-minimal-daily:24.04}"
+LXD_SNAP_CHANNEL="${LXD_SNAP_CHANNEL:-latest/stable}"
+
+# waitNestedInstanceReady: waits for the instance running within another LXD
+# instance to become ready (processes count > 1).
+waitNestedInstanceReady() (
```

Comment:
Can we use the `declare` trick? Or maybe have that function be a wrapper around it?